src/features/identityCheck/pages/profile/SetCity.tsx

```diff
@@ -0,0 +1,140 @@
+import { t } from '@lingui/macro'
+import { useNavigation } from '@react-navigation/native'
+import debounce from 'lodash.debounce'
+import React, { useEffect, useRef, useState } from 'react'
+import { Keyboard, TouchableOpacity } from 'react-native'
+
+import { AddressOption } from 'features/identityCheck/atoms/AddressOption'
+import { CenteredTitle } from 'features/identityCheck/atoms/CenteredTitle'
+import { PageWithHeader } from 'features/identityCheck/components/layout/PageWithHeader'
+import { useIdentityCheckContext } from 'features/identityCheck/context/IdentityCheckContextProvider'
+import { IdentityCheckError } from 'features/identityCheck/errors'
+import { UseNavigationType } from 'features/navigation/RootNavigator'
+import { eventMonitoring } from 'libs/monitoring'
+import { SuggestedCity } from 'libs/place'
+import { useCities } from 'libs/place/useCities'
+import { accessibilityAndTestId } from 'tests/utils'
+import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
+import { InputError } from 'ui/components/inputs/InputError'
+import { TextInput } from 'ui/components/inputs/TextInput'
+import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
+import { Invalidate } from 'ui/svg/icons/Invalidate'
+import { Spacer } from 'ui/theme'
+import { ACTIVE_OPACITY } from 'ui/theme/colors'
+
+export const SetCity = () => {
+  const { showErrorSnackBar } = useSnackBarContext()
+  const { navigate } = useNavigation<UseNavigationType>()
+  const { dispatch } = useIdentityCheckContext()
+  const [query, setQuery] = useState('')
+  const [debouncedPostalCode, setDebouncedPostalCode] = useState<string>(query)
+  const [selectedCity, setSelectedCity] = useState<SuggestedCity | null>(null)
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
+
+  const debouncedSetCity = useRef(debounce(setDebouncedPostalCode, 500)).current
+  const { data: cities = [], isError, isSuccess } = useCities(debouncedPostalCode)
+
+  useEffect(() => {
+    if (!isError) return
+
+    showErrorSnackBar({
+      message: t`Nous avons eu un problème pour trouver la ville associée à ton code postal. Réessaie plus tard.`,
+      timeout: SNACK_BAR_TIME_OUT,
+    })
+
+    eventMonitoring.captureException(
+      new IdentityCheckError('Failed to fetch data from API: https://geo.api.gouv.fr/communes')
+    )
+  }, [isError])
+
+  useEffect(() => {
+    if (isSuccess && cities.length === 0)
+      setErrorMessage(
+        t`Ce code postal est introuvable. Réessaye un autre code postal ou renseigne un arrondissement (ex: 75001).`
+      )
+  }, [isSuccess, cities.length])
+
+  const onChangePostalCode = (value: string) => {
+    setQuery(value)
+    debouncedSetCity(value)
+    setSelectedCity(null)
+    if (query.length < 5) setErrorMessage(null)
```

Comment:
```suggestion
    setErrorMessage(null)
```

ça dépend pas de `query.length`